rancher2/resource_rancher2_machine_config_v2.go

```diff
@@ -46,9 +46,28 @@ func resourceRancher2MachineConfigV2Create(d *schema.ResourceData, meta interfac
 		Delay:      1 * time.Second,
 		MinTimeout: 3 * time.Second,
 	}
-	_, waitErr := stateConf.WaitForState()
-	if waitErr != nil {
-		return fmt.Errorf("[ERROR] waiting for machine config (%s) to be active: %s", newObj.ID, waitErr)
+
+	// Terraform may finish creating the machine config v2 before rancher has set get
+	// permissions on the resource. Retry a few times to see if the state will return
+	// without error.
+	var machineConfigErrorState bool
+
+	for i := 0; i < 4; i++ {
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			machineConfigErrorState = true
+			if IsNotAccessibleByID(err) {
+				log.Printf("[INFO] Retrying state for Machine Config V2 %s", newObj.ID)
+				continue
+			}
+		}
+		return flattenMachineConfigV2(d, newObj)
+	}
+
+	// If we retry four times and the machine config v2 is still inaccessible or in an
+	// error state, return err.
+	if machineConfigErrorState {
```

Comment:
Currently, this check is not needed because, if we reach this point, then `machineConfigErrorState` is `true`.